src/main/webapp/app/overview/course-statistics/course-statistics.component.ts

```diff
@@ -342,132 +330,27 @@ export class CourseStatisticsComponent implements OnInit, OnDestroy {
-            }
-
-            return;
-        }
-
-        // exercise results strings are mostly 'x points' or 'x of y points'
-        if (replaced.includes('points')) {
-            if (split.length === 2) {
-                groupedExercise.scores.tooltips.push(
-                    this.translateService.instant('artemisApp.courseOverview.statistics.exerciseAchievedScore', {
-                        points: parseFloat(split[0]),
-                        percentage: roundedScore,
-                    }) + this.generateTooltipExtension(includedInOverallScore),
-                );
-                groupedExercise.missedScores.tooltips.push(
-                    this.translateService.instant('artemisApp.courseOverview.statistics.exerciseMissedScore', {
-                        points: '',
-                        percentage: 100 - score,
-                    }),
-                );
-                return;
-            }
-            if (split.length === 4) {
-                groupedExercise.scores.tooltips.push(
-                    this.translateService.instant('artemisApp.courseOverview.statistics.exerciseAchievedScore', {
-                        points: parseFloat(split[0]),
-                        percentage: roundedScore,
-                    }) + this.generateTooltipExtension(includedInOverallScore),
-                );
-                groupedExercise.missedScores.tooltips.push(
-                    this.translateService.instant('artemisApp.courseOverview.statistics.exerciseMissedScore', {
-                        points: missedPoints,
-                        percentage: 100 - score,
-                    }),
-                );
-                return;
-            }
-        }
-
-        // programming exercise result strings are mostly 'x passed' or 'x of y passed'
-        if (replaced.includes('passed')) {
-            if (split.length === 2) {
-                groupedExercise.scores.tooltips.push(parseFloat(split[0]) + ' tests passed (' + roundedScore + '%).' + this.generateTooltipExtension(includedInOverallScore));
-                groupedExercise.missedScores.tooltips.push('(' + (100 - score) + '%)');
-                return;
-            }
-            if (split.length === 4) {
-                groupedExercise.scores.tooltips.push(parseFloat(split[0]) + ' tests passed (' + roundedScore + '%).' + this.generateTooltipExtension(includedInOverallScore));
-                groupedExercise.missedScores.tooltips.push(missedPoints + ' tests failed (' + (100 - score) + '%).');
-                return;
-            }
-        }
+    /**
+     * Generates array containing default configuration for every possible part in one stacked bar
+     * @private
+     * @returns dedicated object that is requested by ngx-charts in order to visualize one bar in the horizontal bar chart
+     */
+    private static generateDefaultSeries(): any[] {
+        return [
+            { name: 'No due date', value: 0, absoluteValue: 0 },
```

Comment:
See my thoughts about enumerating these values; setting this comment so we don't forget these here as well in case we do it